test/hotspot/jtreg/vmTestbase/nsk/monitoring/MemoryNotificationInfo/from/from001.java

```diff
@@ -133,41 +140,58 @@ public boolean continueExecution() {
+        int oomCount = GarbageUtils.eatMemory(stresser);
+        log.display("eatMemory returns OOM count: " + oomCount);
+
+        // Check for the message.  Poll on queue to avoid waiting forver on failure.
+        // Notification is known to fail, very rarely, with -Xcomp where the allocations
+        // do not affect the monitored pool. Possibly a timing issue, where the "eatMemory"
+        // is done before Notification/threshold processing happens.
+        // The Notification is quite immediate, other than that problem.
+        boolean messageReceived = false;
+        int tries = 0;
+        while (!messageReceived && ++tries < MAX_TRIES) {
             try {
-                from001Listener.queue.take();
-                messageNotRecieved = false;
+                Object r = from001Listener.queue.poll(10000, TimeUnit.MILLISECONDS);
+                if (r == null) {
+                    log.display("poll for Notification data returns null...");
+                    continue;
+                } else {
+                    messageReceived = true;
+                    break;
+                }
             } catch (InterruptedException e) {
-                messageNotRecieved = true;
+                // ignored, continue
             }
         }
 
-        result = MemoryNotificationInfo.from(from001Listener.data.get());
-        try {
-           ObjectName poolObjectName = new ObjectName(monitor.getName(poolObject));
-           ObjectName resultObjectName = new ObjectName(
-                     ManagementFactory.MEMORY_POOL_MXBEAN_DOMAIN_TYPE +
-                     ",name=" + result.getPoolName());
-
-           log.display("poolObjectName : " + poolObjectName +
-                              " resultObjectName : " + resultObjectName);
-
-           if (!poolObjectName.equals(resultObjectName)) {
-              log.complain("FAILURE 3.");
-              log.complain("Wrong pool name : " + resultObjectName +
-                           ", expected : " + poolObjectName);
-              testFailed = true;
-           }
-
-        } catch (Exception e) {
-            log.complain("Unexpected exception " + e);
-            e.printStackTrace(log.getOutStream());
-            testFailed = true;
+        // If we got a Notification, test that the CompositeData can create a MemoryNotificationInfo
+        if (!messageReceived) {
+            throw new TestFailure("No Notification received.");
+        } else {
+            result = MemoryNotificationInfo.from(from001Listener.data.get());
+            try {
+               ObjectName poolObjectName = new ObjectName(monitor.getName(poolObject));
+               ObjectName resultObjectName = new ObjectName(
+                         ManagementFactory.MEMORY_POOL_MXBEAN_DOMAIN_TYPE +
+                         ",name=" + result.getPoolName());
+
+               log.display("poolObjectName : " + poolObjectName +
+                                  " resultObjectName : " + resultObjectName);
```

Comment:
Nit: It'd be nice to fix the incorrect indent (it was in original code).

Reply:
right, have made a formatting update